core/os/os.cpp

```diff
@@ -553,6 +553,16 @@ void OS::add_frame_delay(bool p_can_draw) {
 	}
 }
 
+OS::PreferredTextureFormat OS::get_preferred_texture_format() const {
+#if defined(__arm__) || defined(__aarch64__) || defined(_M_ARM) || defined(_M_ARM64)
+	return PREFERRED_TEXTURE_FORMAT_ETC2_ASTC; // By rule, ARM hardware uses ETC texture compression.
+#elif defined(__x86_64__) || defined(_M_X64) || defined(i386) || defined(__i386__) || defined(__i386) || defined(_M_IX86)
+	return PREFERRED_TEXTURE_FORMAT_S3TC_BPTC; // By rule, X86 hardware prefers S3TC and derivatives.
+#else
+	return PREFERRED_TEXTURE_FORMAT_S3TC_BPTC; // Override in platform if needed.
+#endif
+}
+
```

Comment:
This probably will need an extra check for Rosetta on macOS, I guess override like:

```c++
OS::PreferredTextureFormat OS_MacOS::get_preferred_texture_format() const {
#if defined(__aarch64__)
    return PREFERRED_TEXTURE_FORMAT_ETC2_ASTC; // By rule, ARM hardware uses ETC texture compression.
#elif defined(__x86_64__)
    int32_t ret = 0;
    size_t ret_size = sizeof(ret);
    if (sysctlbyname("sysctl.proc_translated", &ret, &ret_size, nulptr, 0) == 0) {
        if (ret == 1) {
            return PREFERRED_TEXTURE_FORMAT_ETC2_ASTC; // X86 binary running on ARM hardware.
        }
    }
    return PREFERRED_TEXTURE_FORMAT_S3TC_BPTC; // By rule, X86 hardware prefers S3TC and derivatives.
#endif
}
```

Also, is it true for the other ARM platforms (Android, Windows for ARM)?

Reply:
For Windows on ARM I suppose its the same thing, it uses Android hardware..